### PR TITLE
Chatlog: Support viewing log stats for a room

### PR DESCRIPTION
**Pull request by mia-pi-git**
Useful feature from several bots that PS can do better.

### CODE REVIEW

**Comment by Zarel** [comment]
I think peach said something about `rg -C` or `rg -c` or something? Should I wait for that change?

**Comment by mia-pi-git** [comment]
> I think peach said something about `rg -C` or `rg -c` or something? Should I wait for that change?

I pushed that already. See line 701.

File: server/chat-plugins/chatlog.ts [hunk 1]
```diff
@@ -666,17 +687,135 @@ export const LogSearcher = new class {
...
+			buf += `<small><a roomid="view-roomstats-${roomid}--${prevMonth}${user ? `--${user}` : ''}">Previous month</a></small>`;
+		}
+		if (FS(`logs/chat/${roomid}/${nextMonth}`).existsSync()) {
+			buf += ` <small><a roomid="view-roomstats-${roomid}--${nextMonth}${user ? `--${user}` : ''}">Next month</a></small>`;
+		}
+		buf += `<hr /><ol>`;
+		if (user) {
+			const sortedDays = Object.keys(results).sort((a, b) => {
+				return new Date(b).getTime() - new Date(a).getTime();
+			});
```
Comment (Zarel) [inline server/chat-plugins/chatlog.ts:771]:
```suggestion
			const sortedDays = Object.keys(results).sort((a, b) => (
				new Date(b).getTime() - new Date(a).getTime()
			));
```

I feel like I've suggested this a lot lately.

File: server/chat-plugins/chatlog.ts [hunk 2]
```diff
@@ -666,17 +687,135 @@ export const LogSearcher = new class {
...
+			for (const date in results) {
+				for (const userid in results[date]) {
+					if (!totalResults[userid]) totalResults[userid] = 0;
+					totalResults[userid] += results[date][userid];
+				}
+			}
+			const resultKeys = Object.keys(totalResults);
+			const sortedResults = resultKeys.sort((a, b) => {
+				return totalResults[b] - totalResults[a];
+			}).slice(0, MAX_TOPUSERS);
```
Comment (Zarel) [inline server/chat-plugins/chatlog.ts:788]:
```suggestion
			const sortedResults = resultKeys.sort((a, b) => (
				totalResults[b] - totalResults[a]
			)).slice(0, MAX_TOPUSERS);
```

**Comment by Zarel** [comment]
https://eslint.org/docs/rules/arrow-body-style#require-braces-in-arrow-function-body-arrow-body-style

We should be turning this on, apparently.

**Comment by Zarel** [comment]
Thanks. The `resultSep` stuff is kinda confusing but it seems to be working around a ripgrep quirk?

**Comment by mia-pi-git** [comment]
> Thanks. The `resultSep` stuff is kinda confusing but it seems to be working around a ripgrep quirk?

Yeah, it's not my favorite but ripgrep uses different separators for searches with -m as opposed to ones without.

**Comment by mia-pi-git** [comment]
It uses `---` to deliminate context blocks.

**Pull request merged**